flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -338,6 +339,6 @@ protected FlinkKubeClient createFlinkKubeClient() {
 
 	@Override
 	protected double getCpuCores(Configuration configuration) {
-		return flinkConfig.getDouble(KubernetesConfigOptions.TASK_MANAGER_CPU, numSlotsPerTaskManager);
+		return TaskExecutorProcessUtils.getCpuCoresWithFallbackConfigOption(configuration, KubernetesConfigOptions.TASK_MANAGER_CPU);
```

Comment:
weird that `ActiveResourceManager#getCpuCores` returns double, though we need `CPUResource` which is in general returned from `getCpuCoresWithFallback`. we need int rewrapping only in yarn.
This could also eliminate `TaskExecutorProcessSpecBuilder#withCpuCores(double cpuCores)`.

Reply:
True. It would be better to warp `CPUResource` inside `ActiveResourceManager#getCpuCores`.